### ISSUE TITLE
Building an RPM - RHEL 7.3

### ISSUE THREAD

**Issue by kesterriley** [apple/cups#5032]
Hi,

I am trying to build an RPM on Rhel 7.3.

I have tried:

 `rpmbuild -ta cups-2.2.0-source.tar.gz --without libusb1 --without dnssd`

and

`rpmbuild -ta cups-2.1.4-source.tar.gz --without libusb1 --without dnssd`

but on both the rpmbuild process errors out with:

```
error: File not found: /root/rpmbuild/BUILDROOT/cups-2.1.4-0.x86_64/usr/bin/ippfind
error: File not found: /root/rpmbuild/BUILDROOT/cups-2.1.4-0.x86_64/usr/share/man/man1/ippfind.1.gz


```

Any one able to help me??

Thanks

**Comment by imsedgar** [comment]
You may try to use one of the Fedora source packages,  or at least use the .spec file as a starting point (may need modifications) with a newer source version from github (as tar.gz). But I need to say that I didn't build cups on RHEL, and I didn't compared file  packaging/cups.spec.in from github with cups.spec from Fedora, because I always used that of Fedora (for Fedora...).
https://koji.fedoraproject.org/koji/packageinfo?packageID=54
https://koji.fedoraproject.org/koji/buildinfo?buildID=888069
...

**Comment by kesterriley** [comment]
OK thanks I shall try that

**Comment by michaelrsweet** [comment]
Looks like we need to add ippfind to the list of programs that doesn't get installed without dnssd support (Avahi).

(but I will note that CUPS is just about useless without Avahi on Linux unless you just want to print to USB printers)

**Comment by kesterriley** [comment]
@michaelrsweet  - thansk for your reply.

Initially I was not trying to exclude it (Avahi), but avahi-devel is was not available (at least to me in my channels on RHEL 7.3). I have avahi installed at version 0.6.31-17.el7.

 `rpmbuild -ta cups-2.2.0-source.tar.gz --without libusb1`

```
error: Failed build dependencies:
        avahi-devel is needed by cups-1:2.2.0-0.x86_64
```

Since then I have found a avahi-devel package for Red Hat 7, and when I try building the RPM again I have a load of other dependency miss matches.

`Processing files: cups-2.2.0-0.x86_64
Provides: LPRng application() application(cups.desktop) config(cups) = 1:2.2.0-0 cups = 1:2.2.0-0 cups(x86-64) = 1:2.2.0-0 lpd lpr
Requires(interp): /bin/sh /bin/sh /bin/sh
Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
Requires(post): /bin/sh
Requires(preun): /bin/sh
Requires(postun): /bin/sh
Requires: libavahi-client.so.3()(64bit) libavahi-common.so.3()(64bit) libc.so.6()(64bit) libc.so.6(GLIBC_2.14)(64bit) libc.so.6(GLIBC_2.15)(64bit) libc.so.6(GLIBC_2.16)(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libc.so.6(GLIBC_2.3)(64bit) libc.so.6(GLIBC_2.3.2)(64bit) libc.so.6(GLIBC_2.3.4)(64bit) libc.so.6(GLIBC_2.4)(64bit) libcrypt.so.1()(64bit) libcups.so.2()(64bit) libcupscgi.so.1()(64bit) libcupsimage.so.2()(64bit) libcupsmime.so.1()(64bit) libcupsppdc.so.1()(64bit) libdbus-1.so.3()(64bit) libdl.so.2()(64bit) libgcc_s.so.1()(64bit) libgcc_s.so.1(GCC_3.0)(64bit) libgnutls.so.28()(64bit) libm.so.6()(64bit) libm.so.6(GLIBC_2.2.5)(64bit) libpam.so.0()(64bit) libpam.so.0(LIBPAM_1.0)(64bit) libpthread.so.0()(64bit) libpthread.so.0(GLIBC_2.2.5)(64bit) libstdc++.so.6()(64bit) libstdc++.so.6(CXXABI_1.3)(64bit) libstdc++.so.6(GLIBCXX_3.4)(64bit) libsystemd.so.0()(64bit) libsystemd.so.0(LIBSYSTEMD_209)(64bit) libz.so.1()(64bit) rtld(GNU_HASH)
Obsoletes: lpd lpr LPRng cups-da cups-de cups-es cups-et cups-fi cups-fr cups-he cups-id cups-it cups-ja cups-ko cups-nl cups-no cups-pl cups-pt cups-ru cups-sv cups-zh
Processing files: cups-devel-2.2.0-0.x86_64
Provides: cups-devel = 1:2.2.0-0 cups-devel(x86-64) = 1:2.2.0-0
Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
Requires: /bin/sh libavahi-client.so.3()(64bit) libavahi-common.so.3()(64bit) libc.so.6()(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libc.so.6(GLIBC_2.3)(64bit) libc.so.6(GLIBC_2.3.4)(64bit) libc.so.6(GLIBC_2.4)(64bit) libcrypt.so.1()(64bit) libcups.so.2()(64bit) libcupscgi.so.1()(64bit) libcupsimage.so.2()(64bit) libcupsmime.so.1()(64bit) libcupsppdc.so.1()(64bit) libgcc_s.so.1()(64bit) libgcc_s.so.1(GCC_3.0)(64bit) libm.so.6()(64bit) libpthread.so.0()(64bit) libpthread.so.0(GLIBC_2.2.5)(64bit) libstdc++.so.6()(64bit) libstdc++.so.6(CXXABI_1.3)(64bit) libstdc++.so.6(GLIBCXX_3.4)(64bit) libz.so.1()(64bit) rtld(GNU_HASH)
Processing files: cups-libs-2.2.0-0.x86_64
Provides: cups-libs = 1:2.2.0-0 cups-libs(x86-64) = 1:2.2.0-0 libcups.so.2()(64bit) libcups1 libcupscgi.so.1()(64bit) libcupsimage.so.2()(64bit) libcupsmime.so.1()(64bit) libcupsppdc.so.1()(64bit)
Requires(interp): /bin/sh /bin/sh
Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
Requires(post): /bin/sh
Requires(postun): /bin/sh
Requires: libavahi-client.so.3()(64bit) libavahi-common.so.3()(64bit) libc.so.6()(64bit) libc.so.6(GLIBC_2.14)(64bit) libc.so.6(GLIBC_2.16)(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libc.so.6(GLIBC_2.3)(64bit) libc.so.6(GLIBC_2.3.4)(64bit) libc.so.6(GLIBC_2.4)(64bit) libcrypt.so.1()(64bit) libcups.so.2()(64bit) libgcc_s.so.1()(64bit) libgcc_s.so.1(GCC_3.0)(64bit) libgnutls.so.28()(64bit) libgnutls.so.28(GNUTLS_1_4)(64bit) libgnutls.so.28(GNUTLS_2_10)(64bit) libgnutls.so.28(GNUTLS_3_0_0)(64bit) libm.so.6()(64bit) libm.so.6(GLIBC_2.2.5)(64bit) libpthread.so.0()(64bit) libpthread.so.0(GLIBC_2.2.5)(64bit) libpthread.so.0(GLIBC_2.3.2)(64bit) libstdc++.so.6()(64bit) libstdc++.so.6(CXXABI_1.3)(64bit) libstdc++.so.6(GLIBCXX_3.4)(64bit) libz.so.1()(64bit) libz.so.1(ZLIB_1.2.0)(64bit) rtld(GNU_HASH)
Processing files: cups-lpd-2.2.0-0.x86_64
Provides: cups-lpd = 1:2.2.0-0 cups-lpd(x86-64) = 1:2.2.0-0
Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
Requires: libavahi-client.so.3()(64bit) libavahi-common.so.3()(64bit) libc.so.6()(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libc.so.6(GLIBC_2.3)(64bit) libc.so.6(GLIBC_2.3.4)(64bit) libc.so.6(GLIBC_2.4)(64bit) libc.so.6(GLIBC_2.7)(64bit) libcrypt.so.1()(64bit) libcups.so.2()(64bit) libm.so.6()(64bit) libpthread.so.0()(64bit) libpthread.so.0(GLIBC_2.2.5)(64bit) libz.so.1()(64bit) rtld(GNU_HASH)
Processing files: cups-debuginfo-2.2.0-0.x86_64
Provides: cups-debuginfo = 1:2.2.0-0 cups-debuginfo(x86-64) = 1:2.2.0-0
Requires(rpmlib): rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1 rpmlib(CompressedFileNames) <= 3.0.4-1
Checking for unpackaged file(s): /usr/lib/rpm/check-files /root/rpmbuild/BUILDROOT/cups-2.2.0-0.x86_64
error: Installed (but unpackaged) file(s) found:
   /usr/share/doc/cups/pt_BR/index.html
   /usr/share/locale/pt_BR/cups_pt_BR.po


RPM build errors:
    Installed (but unpackaged) file(s) found:
   /usr/share/doc/cups/pt_BR/index.html
   /usr/share/locale/pt_BR/cups_pt_BR.po
`

Basically, I am trying to get the latest version of cups, to work on the latest version of RHEL! We have various printer issues with new Canon printers, and they wont provide support unless we are running the latest version of CUPS, and not the 1.4 version we currently run on RHEL 6.

**Comment by michaelrsweet** [comment]
The latest version of CUPS is 2.2.3. I believe 2.2.0 did have an issue of the Portuguese localization not being included...

**Comment by kesterriley** [comment]
@michaelrsweet OK thanks, I had downloaded lots of different versions to try - that one does build OK. I had not tried it since I installed the avahi-devel package. 

The initial problem still stands though - thanks for your help

**Comment by kesterriley** [comment]
@michaelrsweet I have found another issue as well I am afraid. I have not used the --without system switch, but when it tries to install the RPM, it is using /sbin/service.

I have extracted the scripts from the newly created RPM and it is running:

/sbin/chkconfig --add cups
....
/sbin/service cups stop.

So, on install I get an error "error reading information on service cups: No Such file or directory", and then obviously then the service is not there for me to stop or start.
